src/Avalonia.Controls/Platform/Win32CornerHints.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+
+namespace Avalonia.Platform;
+
+/// <summary>
+/// Specifies hints for window corner appearance.
+/// </summary>
+[Flags]
+public enum Win32WindowCornerHints
+{
+    /// <summary>
+    /// Default Avalonia behavior.
+    /// </summary>
+    NoHint,
+
+    /// <summary>
+    /// The platform's default corner style.
+    /// </summary>
+    PlatformDefault,
```

Comment:
`NoHint` is exactly the same as `PlatformDefault`. We shouldn't have two values here. These should be combined into a `Default` value. `Default` means no hint is given so the platform defaults will be used.

Reply:
This is the current behavior:
https://github.com/AvaloniaUI/Avalonia/blob/0bd90d802a208e4d4247d188a3fadbe129087f2e/src/Windows/Avalonia.Win32/WindowImpl.cs#L1150-L1174

So it is `DWMWCP_ROUND` when `_isClientAreaExtended` is true and window state is not full screen. Otherwise it uses `DWMWCP_DEFAULT`.

This is because DWMWCP_DEFAULT with client area extended results in not rounded corners.

This is also why I've split the enum into `NoHint` (aka avalonia default) and `Default` (aka platform default). However, I could agree than maybe `Default` aka "Platform Default" is not useful in practice. Avalonia default is good for 99.9% programmers and for the small percentage the choice between "rounded" and "not rounded" it probably enough.

Reply:
Reading, I think we need the following values: Default, Rounded, RoundedSmall, NotRounded (Alternative: Round, RoundSmall, DoNotRound following upstream API)